internal/provider/resource_hcp_peering_connection_activation.go

```diff
@@ -0,0 +1,118 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var activationDefaultTimeout = time.Minute * 5
+
+// The purpose of this resource is to ensure the given peering connection is active before the Terraform run returns. Since
+// peering connections generally must return early with some outputs in order to complete the peering, it may take some time before the peering connection is active.
+// This resource polls the peering connection and returns once it verifies the state is active.
+func resourcePeeringConnectionActivation() *schema.Resource {
```

Comment:
On second thought, this might make more sense as a read-only [data source](https://www.terraform.io/language/data-sources) 🤔

Reply:
Yea, that's what I was thinking too when reading through. I think a data source might fit better, and probably makes more sense since we don't appear to actually be creating a piece of infrastructure.

Reply:
+1 Data source seems to be the better fit, yeah.